src/main/java/org/opentripplanner/transit/raptor/api/request/RaptorRequestBuilder.java

```diff
@@ -119,4 +125,36 @@ public DebugRequestBuilder debug() {
   public RaptorRequest<T> build() {
     return new RaptorRequest<>(this);
   }
+
+  /**
+   * Generate a name to the RaptorRouting request that can be used in debugging, logging
+   * and/or performance monitoring.
+   * <p>
+   * Note! The {@code profile}, {@code searchDirection}, {@code optimizations} is used to
```

Comment:
It will throw an NPE if not set

Reply:
Actually, the fields are not null - they have a default values. The alias is needed before the builder is complete to set the name of the timers, and the timers is added to the builder. I enforced this by using a flag inside the builder. An exception is thrown, if one of the fields in the alias is modified after the alias is generated. This will restrict the builder a bit, not allowing to use the builder in tests to generate more than one request, but we can always add a `copy() : Builder` method to the request class if that is needed.